[KLinkesV] iP

LinkesV opened the pull request:
## Cypher Chat Bot 

> Any fool can write code that a computer can understand. Good programmers write code that humans can understand. --Martin Fowler [[source](http://www.amazon.com/exec/obidos/redirect?link_code=as2&path=ASIN/0201485672&tag=anothedayinmy-20&camp=1789&creative=9325)]

### Why use **Cypher**?:
- Easy to learn
- Use CLI for all commands
- ~~Fast~~ Faster to use than other ChatBots out there

Features of Cypher:
- [x] Manage different types of task such as Todo, Deadlines and Events
- [ ]  Mark task as complete
- [ ] Store previous session data in a local file
- [x] Easy to use text-based commands

**Download** and use now by follow these steps:
1. download it from here
2. run the jar file in terminal
3. and you are good to go 

Here is some sample code from this project:
```
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}

```





.

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

LinkesV committed:
Level 0 (Rename, Greet, Exit)

LinkesV committed:
Level 1 (Echo)

LinkesV committed:
Level 2 (Add, list)

LinkesV committed:
Level 3 (Mark, UnMark)

LinkesV committed:
Level 4 (ToDos, Events, Deadlines)

LinkesV committed:
Automated Text UI Testing (Basic commands tested in the input file)

LinkesV committed:
Level 5 (Handle Errors)

LinkesV committed:
Level 6 (Delete)

LinkesV committed:
Addition of enums and switch statements

LinkesV committed:
Create new branch for level-7 feature

LinkesV committed:
Save tasks to the hard disk and read them upon Cypher starting up

LinkesV committed:
Chatbot able to understand date and times

LinkesV committed:
Delete tasks from file on harddrive

LinkesV committed:
Merge branch 'branch-Level-7' to the main branch

LinkesV committed:
Update Event output

LinkesV committed:
Merge branch 'branch-Level-8' with the main branch

LinkesV committed:
Abstract commands into different classes

LinkesV committed:
Merge remote-tracking branch 'origin/add-gradle-support' into the main branch

LinkesV committed:
Edit build.gradle file

LinkesV committed:
Build and run Cypher using gradle

LinkesV committed:
Organize classes into Java Packages

LinkesV committed:
Test 2 methods using JUnit

LinkesV committed:
Package app as JAR file

LinkesV committed:
Add Find command

LinkesV committed:
Add JavaDoc comments

LinkesV committed:
Comply with coding standard

LinkesV committed:
Merge branch 'branch-Level-9' to master branch

LinkesV committed:
Merge branch-A-JavaDoc with master branch and resolve conflicts

LinkesV committed:
Merge branch-A-CodingStandard with master branch and resolve conflicts

LinkesV committed:
Update JUnit

LinkesV committed:
Add checkstyle.xml

LinkesV committed:
Fix chekstyle errors

LinkesV committed:
Merge branch 'branch-Checkstyle' to master branch

annabellekk commented on src/main/java/cypherchatbot/Cypher.java:
```diff
@@ -0,0 +1,72 @@
+
+    public void run() {
+        ui.greet();
+        boolean shouldEnd = false;
```
Would it be better to use a name such as 'hasEnded' here instead?

mongj replied:
I think `shouldEnd` works! `hasEnded` implies that the program has already ended which might be counterintuitive.

annabellekk commented on src/main/java/cypherchatbot/command/ByeCommand.java:
```diff
@@ -0,0 +1,16 @@
+        ui.goodBye();
+    }
+
+    public boolean isExit() {
```
Perhaps could change this to 'returnExitStatus' to make it a verb

annabellekk commented on src/main/java/cypherchatbot/command/DeadlineCommand.java:
```diff
@@ -0,0 +1,60 @@
+ * and add it to the task list in the Cypher chat bot application.
+ */
+public class DeadlineCommand extends Command {
+    private String[] command;
```
Can change it to 'commands' instead since it is a collection of String! :)

annabellekk commented on src/main/java/cypherchatbot/task/Task.java:
```diff
@@ -0,0 +1,28 @@
+        return str;
+    }
+
+    public abstract String toStringinFile();
```
Could change the name to 'toStringInFile()' to follow the camelCase convention

mongj replied:
Agreed, remember to change it in other places in the code too :)

annabellekk commented on src/main/java/cypherchatbot/util/CommandReader.java:
```diff
@@ -0,0 +1,101 @@
+            case HELP:
+                return new HelpCommand();
+            case DELETE:
+                int delVal = Integer.parseInt(command[1]) - 1;
```
Perhaps could expand on the name of the variable delVal to improve readability

annabellekk reviewed (commented):
Good implementation overall! Just some comments on minor naming improvements. I like how you abstracted out all the commands, tasks, and util into not just different classes but different packages as well which makes the structure clearer. Good job!!

LinkesV committed:
Add MainWindow.fxml

LinkesV committed:
Add DialogBox.fxml

LinkesV committed:
Merge branch 'branch-GUI' to master branch

LinkesV committed:
Close GUI application on command

LinkesV committed:
Merge branch 'branch-Level-10' with master branch

mongj commented on src/main/java/cypherchatbot/Cypher.java:
```diff
@@ -0,0 +1,92 @@
+     *  Initializes the Cypher class with the preferred file path. Executes the run
+     *  method in order to start up the Cypher Chat Bot Application
+     */
+
```
Could remove the linebreak here else it is not clear that the comment above is meant for the `main` method.

```suggestion

```

mongj commented on src/main/java/cypherchatbot/util/CommandReader.java:
```diff
@@ -0,0 +1,101 @@
+     *
+     * @param input The string input given by the user
+     * @return A Command Object depending on the input given by the user
+     * @throws CypherException Throws a custom exception if an invalid input is given by the user
```
I think it is not necessary to mention "Throws a custom exception" as it is implied. 

```suggestion
     * @throws CypherException If an invalid input is given by the user.
```

I believe this is more closely aligned with the Javadoc conventions as well.

mongj commented on src/main/java/cypherchatbot/util/CommandReader.java:
```diff
@@ -0,0 +1,101 @@
+     * Parses through the given input by user and instantiates a new Command Class through the use
+     * of enums and switch case statements.
+     *
+     *
```
Just one empty line is required

```suggestion
     *
```

mongj commented on src/main/java/cypherchatbot/util/CommandReader.java:
```diff
@@ -0,0 +1,101 @@
+    }
+
+    /**
+     * Parses through the given input by user and instantiates a new Command Class through the use
```
It may not be necessary to mention that the function uses "enums and switch case statements", since the purpose of comments is to help explain your abstraction to people such that they don't have to read the actual code.

Personally, I think it is sufficient to say something like "Parses through the given input by user and instantiates a new Command Class". Just a thought!

mongj commented on src/main/java/cypherchatbot/util/CommandReader.java:
```diff
@@ -0,0 +1,101 @@
+     * of enums and switch case statements.
+     *
+     *
+     * @param input The string input given by the user
```
Remember to add a period after each line, i.e.

```suggestion
     * @param input The string input given by the user.
```

Same for other places

mongj commented on src/main/java/cypherchatbot/task/Task.java:
```diff
@@ -0,0 +1,28 @@
+        this.completed = false;
+    }
+
+    public void completeTask() {
```
Was wondering if it might be cleaner to name this as `complete()` instead? Calling `completeTask` on a `Task` object might be redundant.

mongj commented on src/main/java/cypherchatbot/task/Task.java:
```diff
@@ -0,0 +1,28 @@
+        this.completed = true;
+    }
+
+    public void incompleteTask() {
```
`incompleteTask()` seems a tad awkward for a method name as incomplete is not a verb. Could possibly consider alternatives like `markAsIncomplete()`

mongj commented on src/main/java/cypherchatbot/util/Ui.java:
```diff
@@ -0,0 +1,112 @@
+import javafx.application.Platform;
+import javafx.util.Duration;
+
+/**
```
I like the descriptive class comments throughout the code!

mongj reviewed (commented):
LGTM! Not much to fix wrt coding standards 👍

LinkesV committed:
Modify MainWindow GUI

LinkesV committed:
Add Launcher.java

> Update build.gradle dependencies
> Change entry point into Applicaton from Main to Launcher
> Helps to avoid file path errors

LinkesV committed:
Overhaul of Command System
> Removed unnecessary compositions
> Rewrote Ui class methods

LinkesV committed:
Improve Code Quality

> Read comments given in IP review
> Follow code quiality guideline
> Follow SLAP and DRY principles

LinkesV committed:
Add assertions

LinkesV committed:
Use Java streams

LinkesV committed:
Merge pull request #1 from LinkesV/branch-A-CodeQuality

Improve Code Quality

LinkesV committed:
Resolved merge conflicts with master

LinkesV committed:
Resolved merge conflicts with master

LinkesV committed:
Merge pull request #3 from LinkesV/branch-A-Streams

Use Java streams

LinkesV committed:
Merge branch 'master' into branch-A-Assertions

LinkesV committed:
Merge pull request #2 from LinkesV/branch-A-Assertions

Add assertions

LinkesV committed:
Add SortCommand.java

LinkesV committed:
Add SortCommand.java

LinkesV committed:
Update HelpCommand.java

LinkesV committed:
Merge branch 'branch-BCDextension' to master branch

LinkesV committed:
Add more exception handling

LinkesV committed:
Merge branch 'branch-A-MoreErrorHandling'

LinkesV committed:
Improve GUI

LinkesV committed:
Add user guide

LinkesV committed:
Update user guide

LinkesV committed:
Update sort command

LinkesV committed:
Update Junit test

LinkesV committed:
Update CommandReader.java

LinkesV committed:
Update JUnit

LinkesV committed:
Fix event bug
